x-pack/platform/plugins/shared/embeddable_alerts_table/public/factories/alerts_table_embeddable_factory.tsx

```diff
@@ -42,69 +48,65 @@ export const getAlertsTableEmbeddableFactory = (
     const timeRange = initializeTimeRange(state);
     const titleManager = initializeTitleManager(state);
     const queryLoading$ = new BehaviorSubject<boolean | undefined>(true);
-    const { data, fieldFormats, licensing } = deps;
-    const { http, application, notifications, settings } = core;
+    const services = {
+      ...coreServices,
+      ...deps,
+    };
+    const tableConfig$ = new BehaviorSubject<EmbeddableAlertsTableConfig>(state.tableConfig);
     const api = buildApi(
       {
         ...timeRange.api,
         ...titleManager.api,
         dataLoading$: queryLoading$,
+        isEditingEnabled: () => true,
+        onEdit: async () => {
+          try {
+            const newTableConfig = await openConfigEditor({
+              coreServices,
+              parentApi: api.parentApi as PresentationContainer,
+              initialConfig: tableConfig$.getValue(),
+            });
+            tableConfig$.next(newTableConfig);
+          } catch {
+            // The user closed without saving
+          }
+        },
+        getTypeDisplayName: () => CONFIG_EDITOR_EDIT_TABLE_TITLE,
         serializeState: () => {
           return {
-            rawState: { ...titleManager.serialize(), ...timeRange.serialize() },
+            rawState: {
+              ...titleManager.serialize(),
+              ...timeRange.serialize(),
+              tableConfig: tableConfig$.getValue(),
+            },
           };
         },
       },
-      { ...titleManager.comparators, ...timeRange.comparators }
+      {
```

Comment:
We're in the process of removing the concept of state comparators in [this PR](https://github.com/elastic/kibana/pull/215947). No action items now, but be aware that the above PR may require quite thorough review when it's ready to ensure that all functionality from this PR was properly ported over.

Reply:
Thanks for the heads up!